File: Filtering in MVC 3/[C#]-Filtering in MVC 3/C#/Mvc3Filter/Models/IUserDB.cs
using System.Collections.Generic;

namespace Mvc3Filter.Models {
    public interface IUserDB {
        List<User> GetAllUsers { get; }
        bool UserExists(string UserName);
        void CreateNewUser(User newUser);
        User GetUser(string uid);
        void Remove(string usrName);
        void Update(User userToUpdate);
    }
}

File: Filtering in MVC 3/[VB]-Filtering in MVC 3/VB/Models/IUserDB.vb
Namespace Mvc3Filter.Models
	Public Interface IUserDB
        Property GetAllUsers() As List(Of User)
		Function UserExists(ByVal UserName As String) As Boolean
		Sub CreateNewUser(ByVal newUser As User)
		Function GetUser(ByVal uid As String) As User
		Sub Remove(ByVal usrName As String)
		Sub Update(ByVal userToUpdate As User)
	End Interface
End Namespace
